src/extension.ts

```diff
@@ -103,3 +135,107 @@ export async function activateInternal(context: vscode.ExtensionContext, perfSta
+            // tslint:disable-next-line: no-unsafe-any
+            await extract(downloadFilePath, { dir: projectFilePath });
+            await requestUtils.downloadFile(
+                `https://raw.githubusercontent.com/microsoft/vscode-dev-containers/master/containers/${devContainerName}/.devcontainer/devcontainer.json`,
+                vscode.Uri.joinPath(devContainerFolderPathUri, 'devcontainer.json').fsPath
+            );
+            await requestUtils.downloadFile(
+                `https://raw.githubusercontent.com/microsoft/vscode-dev-containers/master/containers/${devContainerName}/.devcontainer/Dockerfile`,
+                vscode.Uri.joinPath(devContainerFolderPathUri, 'Dockerfile').fsPath
+            );
+
+            vscode.window.showInformationMessage('Restarting VS Code with your function app project');
+            vscode.commands.executeCommand('vscode.openFolder', vscode.Uri.file(projectFilePath));
+        } catch (err) {
+            vscode.window.showErrorMessage('Failed to set up your local project. Please try again.');
+        } finally {
+            vscode.workspace.fs.delete(
+                vscode.Uri.file(toBeDeletedFolderPathUri.fsPath),
+                {
+                    recursive: true,
+                    useTrash: true
+                }
+            );
+        }
+    } else {
+        vscode.window.showErrorMessage('Invalid inputs. Please try again.');
+    }
+}
+
+function getProjectLanguageForLanguage(language: string): ProjectLanguage {
+    switch (language) {
+        case 'powershell':
+            return ProjectLanguage.PowerShell;
+        case 'node':
+            return ProjectLanguage.TypeScript;
+        case 'python':
+            return ProjectLanguage.Python;
+        case 'java8':
+        case 'java11':
+            return ProjectLanguage.Java;
+        case 'dotnetcore2.1':
+        case 'dotnetcore3.1':
+            return ProjectLanguage.CSharp;
+        default:
+            throw new Error(`Language not supported: ${language}`);
+    }
+}
+
+async function setupAzureAccount(account: AzureAccount): Promise<any> {
+    try {
+        await vscode.commands.executeCommand('azure-account.login');
+        return await account.sessions[0].credentials2.getToken();
+    } catch (err) {
+        vscode.window.showErrorMessage('Failed to setup Azure account. Please try again.');
+        return Promise.resolve(undefined);
+    }
+}
+
+async function activateAzureExtension(azureAccountExt: vscode.Extension<AzureAccount> | undefined): Promise<AzureAccount | undefined> {
+    return await azureAccountExt?.activate();
+}
+
+function getAllRequiredInputs(query: string): { resourceId: string | null; defaultHostName: string | null; devContainerName: string | null; language: string | null; downloadAppContent: string | null } {
+    const queryUrl: URL = new URL(`https://functions.azure.com?${query.toLowerCase()}`);
```

Comment:
I would add a note right above this to indicate we have to generate this URL so we can use the parser below.